### DISCUSSION TITLE
How to encode prisma find many result as JSON?

### DISCUSSION THREAD

**Discussion by felinto-dev** [RobertCraigie/prisma-client-py#332] [Q&A] [answered]
Could you give me some advice about what to do here?

![image](https://user-images.githubusercontent.com/32253743/158004397-15b0e7f9-8879-44c4-8bd1-2b5c605dce26.png)

![image](https://user-images.githubusercontent.com/32253743/158004411-dc263bd3-cb09-46c0-bb76-0d09a51fc734.png)

I need to pack and send Prisma query results to third-party API but I always receive an error.

PS: If I print the "downloads" variable I get a downloads list, as expected.

**Comment by RobertCraigie** [comment, marked as answer]
Prisma queries return [pydantic models](https://pydantic-docs.helpmanual.io/usage/models/). If you need to serialise a model to JSON then you simply have to call `.json()`. 

Serialising a list of models is more complicated,  but this should work for you (I'm on mobile and can't confirm at the moment):

```py
from typing import List
from pydantic import BaseModel
from prisma.models import DownloadRequest

class DownloadRequestList(BaseModel):
    __root__: List[DownloadRequest]

data = DownloadRequestList(__root__=downloads)
return data.json()
```

**Reply by felinto-dev** [reply]
Hi! I couldn't understand much less replicate the solution, sorry.
Could you check this out in more detail when you have time? I think it's a common use case. Thank you very much for your help! <3

**Reply by RobertCraigie** [reply]
No worries, I'll show you how you'd use this in your code and hopefully that'll help.

```py
from typing import List
from pydantic import BaseModel
from prisma.models import DownloadRequest

class DownloadRequestList(BaseModel):
    __root__: List[DownloadRequest]

async def upsert_downloads_request():
    downloads = await DownloadRequest.prisma().find_many(take=5)
    model = DownloadRequestList(__root__=downloads)
    requests.post('url', data=model.json())
```

I know it's a bit awkward having to define a class like that but it's the only method I know of for achieving this. You can dynamically create models but I don't know if that would work for this case and I can't test it at the moment. https://pydantic-docs.helpmanual.io/usage/models/#dynamic-model-creation

**Reply by RobertCraigie** [reply]
Also it looks like you're using requests within an async function, this will block synchronously meaning you lose the benefits of using async, if you want an async HTTP client I'd highly recommend HTTPX.

**Reply by felinto-dev** [reply]
WOW, it works! Thank you!
I will check HTTPX.

**Reply by RobertCraigie** [reply]
Glad I could help :)
